File: test4.py
import math

money = (50000,10000,500,100,50,10)

def change_money(have_money, sum_item, money):
  change = have_money-sum_item
  answer = {}
  for i in money:
    many = math.floor(change/i)
    change -= i*many
    answer[f'{i}원'] = f'{many}개'
  print(answer)

change_money(100000,34830,money)

# python f'{number}개' === js `${number}개`

File: test4.js
const money = [50000, 100, 500, 10000, 50, 10];

money.sort((a, b) => {
  return b - a;
}); //내림차순 정렬

// const no_money = {
//   5000: true,
//   1000: true,
// };

const change_money = (have_money, sum_item, money) => {
  let change = have_money - sum_item;
  let answer = {};
  const aa = (number) => {
    let many = Math.floor(change / number);
    change -= number * many;
    answer[`${number}원`] = `${many}개`;
  };
  money.forEach((number) => {
    aa(number);
  });
  console.log(answer);
};

// const change_money = (have_money, sum_item, money) => {
//   let change = have_money - sum_item;
//   let answer = {};
//   money.forEach((value) => {
//     let many = Math.floor(change / value);
//     change -= value * many;
//     answer[value] = many + "개";
//   });
//   console.log(answer);
// };

change_money(100000, 34830, money);
